core/controllers/feedback_test.py

```diff
@@ -336,3 +344,243 @@ def _get_email(n):
+        thread_id_1 = threads[0]['thread_id']
+        thread_id_2 = threads[1]['thread_id']
+        thread_id_3 = threads[2]['thread_id']
+
+        # Accept a suggestion.
+        self._accept_suggestion(thread_id_1, self._return_true, csrf_token)
+        updated_exploration = exp_services.get_exploration_by_id(self.EXP_ID)
+        self.assertEqual(
+            updated_exploration.states['State A'].content[0].to_dict(),
+            feedback_services.get_suggestion(
+                self.EXP_ID, thread_id_1.split('.')[1])['state_content'])
+
+        # Reject a suggestion.
+        self._reject_suggestion(thread_id_2, self._return_true, csrf_token)
+
+        # Get a list of closed threads with suggestion. 
+        response_dict = self.get_json(
+            '%s/%s?list_type=%s&has_suggestion=%s' % ( 
+            feconf.SUGGESTION_LIST_URL_PREFIX, self.EXP_ID, 'closed', 'true'))
+        self.assertEqual(len(response_dict['threads']), 2)
+
+        # Pass invalid action in the URL.
+        with self.swap(rights_manager.Actor, 'can_edit',
+                       self._return_true):
+            response_dict = self.put_json(
+                '%s/%s/%s' % (feconf.SUGGESTION_ACTION_URL_PREFIX,
+                              self.EXP_ID, thread_id_2.split('.')[1]),
+                {'action': u'rejec'}, csrf_token, expect_errors=True,
+                expected_status_int=400)
+        self.assertIn('Invalid action.', response_dict['error'])
+
+        exception_msg = 'Suggestion has already been accepted/rejected.'
+        # Editor tries to accept rejected suggestion.
+        response_dict = self._accept_suggestion(
+            thread_id_2, self._return_true, csrf_token,
+            expect_errors=True, expected_status_int=500)
+        self.assertIn(exception_msg, response_dict['error'])
+
+        # Editor tries to reject accepted suggestion.
+        response_dict = self._reject_suggestion(
+            thread_id_2, self._return_true, csrf_token,
+            expect_errors=True, expected_status_int=500)
+        self.assertIn(exception_msg, response_dict['error'])
+
+        # Different editor tries to accept rejected suggestion.
+        self.logout()
+        self.login(self.OWNER_EMAIL)
+        csrf_token = self.get_csrf_token_from_response(
+            self.testapp.get('/create/%s' % self.EXP_ID))
+        response_dict = self._accept_suggestion(
+            thread_id_2, self._return_true, csrf_token,
+            expect_errors=True, expected_status_int=500)
+        self.assertIn(exception_msg, response_dict['error'])
+
+        # Different editor tries to reject accepted suggestion.
+        response_dict = self._reject_suggestion(
+            thread_id_2, self._return_true, csrf_token,
+            expect_errors=True, expected_status_int=500)
+        self.assertIn(exception_msg, response_dict['error'])
+
+        # User(non editor) tries to accept a suggestion.
+        self.logout()
```

Comment:
Put this after the previous section, above the comment in the previous line:

        self.assertIn(exception_msg, response_dict['error'])
        self.logout()

        # User (non-editor) tries to accept a suggestion.
        self.login(self.VIEWER_EMAIL)